### ISSUE TITLE
provides/aws: aws_elb does not update listeners

### ISSUE THREAD

**Issue by pmoust** [hashicorp/terraform#1120]
repro; create an `aws_elb` resource, then change a listener port, plan will mark the change as 'modify', plan gets executed, yet no change is applied (there is no implementation in terraform to update listeners in said resource).

```
resource "aws_elb" "foofoo" {
    name = "foofoo"
    subnets = ["${aws_instance.barbar.*.subnet_id}"]

    listener {
        instance_port = 8000
        instance_protocol = "http"
        lb_port = 80
        lb_protocol = "http"
    }
    health_check {
        healthy_threshold = 2
        unhealthy_threshold = 5
        timeout = 4
        target = "HTTP:8080/site/ping"
        interval = 5
    }
}
```

aws-sdk reference: https://github.com/hashicorp/aws-sdk-go/blob/master/gen/elb/elb.go#L153-L163

**Comment by pmoust** [comment]
@catsby port it to aws-sdk-go the other day. @svanharmelen marked it to not `ForceNew` on listener change (which is correct) but there is no functionality implemented to actually update the listener based on schema change.

Now, I have another request that is coupled with this Issue report;
I want to be able to define that 'it's ok' for `instance_port` listener to change in a `listener` block. perhaps adding a block called `mutable`, if i set `mutable = true` then the refresh/plan should not mark the `aws_elb` resource to update for a said `listener` block change. It should only update the state.

My use case is a CoreOS fleet cluster running in an AWS autoscaling group behind some internal ELBs for microservices' management. I want to be able to swap instance listener ports during a rolling update to a service's version. I want to allow this change in listener Schema in terraform, by giving the listener block an instruction to be mutable.

@mitchellh does the above make sense? If so I 'll open a separate issue.

**Comment by daemonza** [comment]
Just came across a similar issue myself. Added second listener

```
aws_elb.elb: Modifying...
  listener.#:                             "1" => "2"
  listener.3057123346.instance_port:      "80" => "80"
  listener.3057123346.instance_protocol:  "http" => "http"
  listener.3057123346.lb_port:            "80" => "80"
  listener.3057123346.lb_protocol:        "http" => "http"
  listener.3057123346.ssl_certificate_id: "" => ""
  listener.3985397548.instance_port:      "" => "9000"
  listener.3985397548.instance_protocol:  "" => "http"
  listener.3985397548.lb_port:            "" => "9000"
  listener.3985397548.lb_protocol:        "" => "http"
  listener.3985397548.ssl_certificate_id: "" => ""
 aws_elb.elb: Modifications complete
```

But the change did not happen.

**Comment by patricklucas** [comment]
Just ran into this as well.

**Comment by jwaldrip** [comment]
:+1: This is happening for me as well.

**Comment by mitchellh** [comment]
Fixed by #1619
